web_src/js/components/DashboardRepoList.vue

```diff
@@ -79,6 +79,13 @@
                   <svg-icon name="octicon-archive" :size="16" class-name="gt-ml-2"/>
                 </span>
               </div>
+              <!-- the commit status icon logic is taken from templates/repo/commit_status.tmpl -->
+              <svg-icon v-if="repo.commit_status_state === 'pending'" name="octicon-dot-fill" class-name="commit-status icon text grey" :size="16"/>
+              <svg-icon v-else-if="repo.commit_status_state === 'running'" name="octicon-dot-fill" class-name="commit-status icon text yellow" :size="16"/>
+              <svg-icon v-else-if="repo.commit_status_state === 'success'" name="octicon-check" class-name="commit-status icon text green" :size="16"/>
+              <svg-icon v-else-if="repo.commit_status_state === 'error'" name="gitea-exclamation" class-name="commit-status icon text red" :size="16"/>
+              <svg-icon v-else-if="repo.commit_status_state === 'failure'" name="octicon-x" class-name="commit-status icon text red" :size="16"/>
+              <svg-icon v-else-if="repo.commit_status_state === 'warning'" name="gitea-exclamation" class-name="commit-status icon text yellow" :size="16"/>
```

Comment:
Should we extract a subtemplate perhaps?

Ah wait…
One thing is Vue, the other Go templates.
Hmm…
A custom component instead?

Reply:
Maybe an object lookup? Should run before or during render.

```js
const commitStatusIcons = {
  pending: ['octicon-dot-fill', 'grey'],
}

const [icon, color] = commitStatusIcons[state];
```

Reply:
Added lookup